components/designSystem/SmolAddressInput.tsx

```diff
@@ -100,36 +147,54 @@ export function SmolAddressInput(): ReactElement {
 
 	const onChange = useCallback(
 		(label: string): void => {
+			set_hasTyped(true);
 			currentInput.current = label;
 			actions.abort();
 			actions.execute(label);
 		},
 		[actions]
 	);
 
+	const onSelectItem = useCallback((item: TAddressBookEntry): void => {
+		currentInput.current = item.label || item.ens || toAddress(item.address);
+		currentLabel.current = item.label || item.ens || toAddress(item.address);
+		currentAddress.current = toAddress(item.address);
+		set_value({
+			address: toAddress(item.address),
+			label: item.label || item.ens || toAddress(item.address),
+			isValid: true,
+			source: 'addressBook'
+		});
+	}, []);
+
 	return (
 		<div className={'group relative h-full w-full max-w-[442px] rounded-lg p-[1px]'}>
 			<div
 				className={cl(
 					'absolute inset-0 z-0 rounded-[9px] transition-colors',
 					status === 'loading'
-						? 'borderPulse'
+						? 'bg-neutral-600'
 						: !isFocused && value.error
 						  ? 'bg-red-500'
-						  : isFocused
-						    ? 'bg-neutral-600'
-						    : 'bg-neutral-400'
+						  : isFocused && value.isValid === true && hasTyped
+						    ? 'bg-green-500'
+						    : isFocused && value.isValid === false && hasTyped
+						      ? 'bg-red-500'
+						      : isFocused
+						        ? 'bg-neutral-600'
+						        : 'bg-neutral-200'
```

Comment:
Nested ternaries are really scary, maybe it's better to move this calculations to separate function/util with using of simple `if` operator so it was readable?